front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -0,0 +1,93 @@
+import type { TFunction } from 'i18next';
+import { keyColumn, createTextColumn } from 'react-datasheet-grid';
+
+import { matchPathStepAndOp } from 'modules/pathfinding/utils';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import type { PathStep } from 'reducers/osrdconf/types';
+import { extractHHMMSS } from 'utils/date';
+import { stripDecimalDigits } from 'utils/numbers';
+import { NO_BREAK_SPACE } from 'utils/strings';
+
+import { marginRegExValidation, MarginUnit } from '../consts';
+import { TableType } from '../types';
+import type { PathStepOpPointCorrespondance, PathWaypointRow } from '../types';
+
+// eslint-disable-next-line import/prefer-default-export
+export const formatSuggestedViasToRowVias = (
+  operationalPoints: SuggestedOP[],
+  pathSteps: PathStep[],
+  t: TFunction<'timesStops', undefined>,
+  startTime?: string,
+  tableType?: TableType
+): PathWaypointRow[] => {
+  let formattedOps = [...operationalPoints];
+  const origin = pathSteps[0] as PathStep;
+  if (origin && 'uic' in origin && 'ch' in origin) {
+    const originIndexInOps = operationalPoints.findIndex(
+      (op) => op.uic === origin.uic && op.ch === origin.ch && op.name === origin.name
+    );
+    // If the origin is in the ops and isn't the first operational point, we need to move it to the first position
+    if (originIndexInOps !== (-1 || 0)) {
+      formattedOps = formattedOps.toSpliced(originIndexInOps, 1);
+      formattedOps.unshift(operationalPoints[originIndexInOps]);
+    }
+  }
+  return formattedOps.map((op, i) => {
+    const pathStep = pathSteps.find((step) => matchPathStepAndOp(step, op));
+    const objectToUse = tableType === TableType.Input ? pathStep || op : op;
+    const { arrival, onStopSignal, name, stopFor, theoreticalMargin } = objectToUse;
+
+    const isMarginValid = theoreticalMargin ? marginRegExValidation.test(theoreticalMargin) : true;
+
+    return {
+      ...op,
+      isMarginValid,
+      arrival: i === 0 ? extractHHMMSS(startTime) : arrival,
+      onStopSignal: onStopSignal || false,
+      name: name || t('waypoint', { id: op.opId }),
+      stopFor,
+      theoreticalMargin,
+    };
+  });
+};
+
+export function findNextScheduledPathStep(
```

Comment:
```suggestion
export function findNextPathStep(
```

Reply:
actually it’s returning an operational Point, the next that corresponds to the next pathStep.
perhaps with can keep findNextScheduledOpPoint ?

Reply:
Yes I agree

Reply:
done